packages/styled-components/src/hoc/withTheme.tsx

```diff
@@ -5,7 +5,17 @@ import determineTheme from '../utils/determineTheme';
 import getComponentName from '../utils/getComponentName';
 import hoist from '../utils/hoist';
 
-export default function withTheme<T extends AnyComponent>(Component: T) {
+export default function withTheme<T extends AnyComponent>(
+  Component: T
+): ReturnType<
+  typeof hoist<
+    React.ForwardRefExoticComponent<
+      React.PropsWithoutRef<React.JSX.LibraryManagedAttributes<T, ExecutionProps>> &
+        React.RefAttributes<T>
+    >,
+    T
+  >
+> {
```

Comment:
I'm using ` ReturnType<typeof hoist<...>>` so the amount of typings to kept in sync is minimum. I need to "copy" the type of `WithTheme` though. Not sure there's a better way to achieve this.